lib/net.js

```diff
@@ -906,8 +906,11 @@ Socket.prototype.connect = function(options, cb) {
     if (localPort && !util.isNumber(localPort))
       throw new TypeError('localPort should be a number: ' + localPort);
 
-    if (port <= 0 || port > 65535)
-      throw new RangeError('port should be > 0 and < 65536: ' + port);
+    if (options.port !== undefined && parseInt(options.port, 10) !== port)
```

Comment:
As mentioned in #libuv, `isNumeric` would not handle negative hexadecimal number strings, and thus `isNumeric` would be inconsistent. But we could have:
```
/*
 * This function returns `true` if `value` is a string that represents a valid
 * decimal or hexadecimal non-negative number, false otherwise.
 */
function isNonNegativeNumeric(n) {
  return !isNaN(parseFloat(n)) && isFinite(n) && n >= 0;
}
``` 
and use it like in the following gist: https://gist.github.com/misterdjules/7e4f25997b3805792aac.